packages/myst-cli/src/build/cff.ts

```diff
@@ -0,0 +1,323 @@
+}
+
+function frontmatterToCFF(frontmatter: PageFrontmatter, abstract?: string): CFF {
+  const { first_page, last_page, issue, volume } = frontmatter.biblio ?? {};
+  const license = frontmatter.license?.content ?? frontmatter.license?.code;
+  const contact = frontmatter.authors
+    ?.filter((author) => author.corresponding)
+    .map((author) => authorToCFF(author, frontmatter.affiliations));
+  let dateString: string | undefined;
+  if (frontmatter.date) {
+    const date = new Date(frontmatter.date);
+    dateString = new Date(date.getTime() - date.getTimezoneOffset() * 60 * 1000)
+      .toISOString()
+      .split('T')[0];
+  }
+  return {
+    'cff-version': '1.2.0',
+    message: 'Please cite the following works when using this project.',
+    type: 'article',
+    abstract,
+    title: frontmatter.title,
+    authors: frontmatter.authors?.map((author) => authorToCFF(author, frontmatter.affiliations)),
+    'date-released': dateString,
+    contact: contact?.length ? contact : undefined,
+    copyright: frontmatter.copyright,
+    identifiers: frontmatter.doi
+      ? [
+          {
+            type: 'doi',
+            value: frontmatter.doi,
+          },
+        ]
+      : undefined,
+    editors: frontmatter.editors
+      ?.map((id) => {
+        const editor = frontmatter.contributors?.find((contrib) => contrib.id === id);
+        if (!editor) return undefined;
+        return authorToCFF(editor, frontmatter.affiliations);
+      })
+      .filter((editor): editor is PersonCFF | EntityCFF => !!editor),
+    end: typeof last_page === 'number' ? last_page : undefined,
+    issue: typeof issue === 'number' ? issue : undefined,
+    'issue-title': typeof issue === 'string' ? issue : undefined,
+    journal: frontmatter.venue?.title,
+    keywords: frontmatter.keywords,
+    license: license?.id,
+    'license-url': license?.url,
+    pages:
+      typeof last_page === 'number' && typeof first_page === 'number'
+        ? last_page - first_page + 1
+        : undefined,
+    start: typeof first_page === 'number' ? first_page : undefined,
+    url: frontmatter.source,
+    volume: typeof volume === 'number' ? volume : undefined,
+    'volume-title': typeof volume === 'string' ? volume : undefined,
+    repository: frontmatter?.github,
+  };
+}
+
+export async function runCffExport(
+  session: ISession,
+  sourceFile: string,
+  exportOptions: ExportWithOutput,
+  opts?: ExportFnOptions,
+) {
+  const toc = tic();
+  const { output, articles } = exportOptions;
+  const { clean, projectPath, extraLinkTransformers } = opts ?? {};
+  const article = articles[0];
+  const state = session.store.getState();
+  let frontmatter: PageFrontmatter | undefined;
+  let abstract: string | undefined;
+  if (projectPath && selectors.selectLocalConfigFile(state, projectPath) === sourceFile) {
+    frontmatter = selectors.selectLocalProjectConfig(state, projectPath);
+    const rawAbstract = frontmatter?.parts?.abstract?.join('\n\n');
```

Comment:
This project frontmatter abstract relies on #1396 - until then, this is just `undefined`